warp/Network/Wai/Handler/Warp/MultiMap.hs

```diff
@@ -81,31 +75,18 @@ toList (MultiMap mm) = concatMap snd $ I.toAscList mm
 pruneWith :: MultiMap v
           -> ((FilePath,v) -> IO Bool)
           -> IO (MultiMap v)
-pruneWith (MultiMap mm) action = MultiMap <$> mm'
-  where
-    !mm' = I.fromAscList <$> go (I.toDescList mm) []
-    go []          !acc = return acc
-    go ((h,s):kss) !acc = do
-        rs <- prune action s
-        case rs of
-            [] -> go kss acc
-            _  -> go kss ((h,rs) : acc)
+pruneWith (MultiMap mm) action = I.foldrWithKey
+  (\h s cont acc -> do
```

Comment:
I would like to change this function to a local function after `where`.

Reply:
Can you explain why? Is it a style preference or does it produce a better core?

Reply:
Just a style preference.

Reply:
OK